.github/workflows/release.yml

```diff
@@ -0,0 +1,218 @@
+name: on_pr_edit
+run-name: 'release: ${{ github.ref_name }}'
+
+on:
+  workflow_dispatch:
+    inputs:
+      next-version:
+        type: string
+        description: Semantic version of the new release to publish
+        required: true
+
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  get-current-state:
+    name: Get the current release state
+    runs-on: ubuntu-latest
+    env:
+      NEXT_VERSION: "${{ inputs.next-version }}"
+    outputs:
+      dmg_filename: "Wallet Recovery Wizard-${{ inputs.next-version }}.dmg"
+      exe_filename: "Wallet Recovery Wizard-Setup-${{ inputs.next-version }}.exe"
+      deb_filename: "Wallet Recovery Wizard-Linux-${{ inputs.next-version }}.deb"
+    steps:
+      - name: Verify the target branch
+        if: github.ref != 'refs/heads/master'
+        run: |
+          echo "This workflow should only run from the master branch"
+          exit 1
+
+      - run: |
+          if [[ ${{ env.NEXT_VERSION }} =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo "The format of the next version is valid."
+          else
+            echo "Invalid format for the next version. Please provide a semantic release version number."
+            exit 1
+          fi
+
+      - name: Check if a release already exists for that version
+        run: |
+          current_release=$(
+            curl -L \
+              -H "Accept: application/vnd.github+json" \
+              -H "Authorization: Bearer ${{ github.token }}" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              https://api.github.com/repos/BitGo/wallet-recovery-wizard/releases/tags/v${{ env.NEXT_VERSION }}
+          )
+
+          current_release_name=$(echo "$current_release" | jq ".name")
+          if [[ -n "$current_release_name" ]]; then
+            echo "A release already exists for version ${{ env.NEXT_VERSION }}."
+            exit 1
+          fi
+
+  mac-build:
+    name: Create Mac release
+    runs-on: macos-latest
+    needs:
+      - get-current-state
+    env:
+      DMG_FILENAME: "${{ needs.get-current-state.outputs.dmg_filename }}"
+      NEXT_VERSION: "${{ inputs.next-version }}"
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: .nvmrc
+          cache: 'npm'
+      - run: |
+          sed -i -e 's/0.0.0-placeholder-version/${{ env.NEXT_VERSION }}/g' "package.json"
+          sed -i -e 's/0.0.0-placeholder-version/${{ env.NEXT_VERSION }}/g' "package-lock.json"
+      # TODO we should pin that version in the package-lock
```

Comment:
Can we create a ticket for this TODO?

Reply:
Here it is https://bitgoinc.atlassian.net/browse/VL-1652